src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils.ts

```diff
@@ -0,0 +1,79 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import { adapters, fromAccountId } from '@shapeshiftoss/caip'
+import type { UtxoBaseAdapter, UtxoChainId } from '@shapeshiftoss/chain-adapters'
+import type { ThornodePoolResponse } from '@shapeshiftoss/swapper'
+import axios from 'axios'
+import { getConfig } from 'config'
+import memoize from 'lodash/memoize'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import type { BigNumber, BN } from 'lib/bignumber/bignumber'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { isUtxoAccountId } from 'state/slices/portfolioSlice/utils'
+
+import type { MidgardPoolResponse, ThorchainSaverPositionResponse } from './types'
+
+const THOR_PRECISION = 8
+
+// Memoized on accountId, see lodash docs:
+// "By default, the first argument provided to the memoized function is used as the map cache key."
+export const getAccountAddresses = memoize(async (accountId: AccountId): Promise<string[]> => {
```

Comment:
only issue with memoization here is you get new utxo addresses each time you do a send

Reply:
Safe since we will realistically never go over the 20 limit over the lifecycle of a page